### PR TITLE
Allow overriding arbitrary arguments

### LINKED ISSUE #67: Changing column titles and font sizes
Hi,
Often I find, I have to give a title to my heatmap which is different than the column/row name of my input dataframe (usually more descriptive). ComplexHeatmap documentation suggests the only way to do that is through column_title/row_title/name. 

Trying to change column titles or fonts using: 
_test.hm<- test.data %>%
heatmap(..., column_title = "Descriptive title for heatmap",...)_

OR trying to change title fonts using:

_test.hm<- test.data %>%
heatmap(..., column_names_gp = gpar(fontsize = 20),...)_

results in: 

_Error in (function (matrix, col, name, na_col = "grey", color_space = "LAB",  : 
  formal argument "column_title" matched by multiple actual arguments_

_Error in (function (matrix, col, name, na_col = "grey", color_space = "LAB",  : 
  formal argument "column_names_gp" matched by multiple actual arguments_

I found a temporary workaround:

test.hm@input$column_title<- "Descriptive title for heatmap"

However, this fix is not ideal since you can't include it in the same function call. A likely cause of this could be a hard-coded value in the source code somewhere. Apologies if a simpler workaround for this exists already.

### PR DESCRIPTION
**Pull request by grst**
Closes #67.

### CODE REVIEW

File: R/functions.R [hunk 1]
```diff
@@ -159,16 +159,17 @@ input_heatmap = function(.data,
 	new(
 		"InputHeatmap",
 		data = .data %>% reduce_to_tbl_if_in_class_chain,
-		input = list(
```
Comment (stemangiola) [inline R/functions.R:162]: Thanks. Can you please comment (in the code as well) on what this contraption is practically doing?

How come row_names_gp are not in your addition?

Comment (grst) [reply R/functions.R:162]: Due to the `.homonyms="last"` parameter, additional arguments passed by the user via `...` overwrite the defaults given below (See also `?rlang::dots_list`). 

So if the user specifies any of the parameters you already have, the ones by the user take precedence. If the user doesn't specify anything, the behaviour is unchanged.

Comment (stemangiola) [reply R/functions.R:162]: Quite elegant.

But why 
```
column_title =
...
row_names_go =
```
are not alongside `column_names_gp` parameter?

Comment (grst) [reply R/functions.R:162]: I'm not sure if I get it, but maybe it's just how the diff is visualized? 
I haven't changed the `column_names_gp` line, except for the comma.

Comment (stemangiola) [reply R/functions.R:162]: My bad. Sorry late at night. I will accept if the test pass.

Thanks.

**Pull request merged**